scraper.js

```diff
@@ -0,0 +1,233 @@
+
+function getBlogDetails() {
+
+    // set the main page url
+    url = "https://blog.zairza.in/"
+
+    // scraped data is stored here
+    let scrapContent = []
+
+
+    // make request to the to be scraped website through request
+    request(url, function (error, response, html) {
+        // check if any errors
+
+        if (!error) {
+            // whole scraping code will be inside this block
+
+            let $ = cheerio.load(html);
+
+            // Declare variables to capture:
+            // TODO: SCRAPE: covers, releases, titles, authors
+
+            let title, href, author, release;
+
+            // make constructor function blog
+            const Blog = function (title, href, author, release, cover) {
+                this.title = title
+                this.href = href
+                this.author = author
+                this.release = release
+            }
+
+            $(".postArticle--short").filter(function () {
+
+                let i = 1
+
+                while (i <= 1) {
+                    title = $(this).find("h3").text()
+                    href = $(this).find(".u-clearfix").next("a").attr("href")
+                    author = $(this).find(".postMetaInline-authorLockup").children().first().text()
+                    release = $(this).find("time").attr("datetime")
+
+                    const newBlog = new Blog(title, href, author, release)
+
+                    scrapContent.push({
+                        "title": newBlog.title,
+                        "href": newBlog.href,
+                        "author": newBlog.author,
+                        "release": newBlog.release
+                    })
+
+                    // continuation of loop
+                    i += 1
+                }
+            })
+
+            // This scrapes the one latest blog post
+            $(".u-paddingTop30").filter(function () {
+
+                title = $(this).find("h3").first().text()
+                href = $(this).find(".u-borderLighter").attr("href")
+                author = $(this).find(".u-flexCenter").children().last().children().first().text()
+                release = $(this).find("time").attr("datetime")
+
+                scrapContent.unshift({
+                    "title": title,
+                    "href": href,
+                    "author": author,
+                    "release": release
+                })
+            })
+
+            // Add data array into a json file
+            fs.writeFile(__dirname + "/json/data.json", JSON.stringify(scrapContent, null, 4), function (err) {
+                if (err) {
+                    console.log(err)
+                }
+            });
+        }
+
+        // scrapContent has all blog objects inside an array
+        // TODO: Add cover image urls into cover.json
+
+        // MAKE A CONSTRUCTION FUNCTION TO STORE COVER URLS
+        const Cover = function (cover) {
+            this.cover = cover
+        }
+
+        // FIRST COVER:
```

Comment:
Why are you fetching images separately for all the posts?

Reply:
Hey, while fetching them using a loop statement they were pushed to the array out of order which made different covers to land on different posts and it was random every time. Is there any better way out? 
**PS**: I also don't support hard-coding fetches separately.

Reply:
Yes, read from the json array you make first. Then after you fetch the image, append it within the json object itself.